lib/iris/analysis/__init__.py

```diff
@@ -1662,6 +1721,37 @@ def interp_order(length):
 """
 
 
+MAX_RUN = Aggregator(
+    None,
+    iris._lazy_data.non_lazy(_lazy_max_run),
+    units_func=lambda units: 1,
```

Comment:
Do you think there's a case that the units here ought to be the same as the coordinate you are collapsing over? If, as in the example, you are calculating the number of consecutive days that a condition holds, it seems like the correct unit ought to be in days.

I imagine that the such behaviour would be complicated to implement, depending on if the coord had bounds or if all the bounds were of uniform length. In the case where the bounds are not uniform length, this complicates what the correct mathematics ought to be. Given this complexity, I can see the appeal of a simpler more robust approach as taken here. Indeed, it seems as though the percintile aggregators also take a simpler approach which doesn't take into account unequal bound width in calculation. There may in fact be cases where you'd want to treat cells with different bound widths with the same weight. In this case a dimensionless unit may actually be the most appropriate one. It's probably worth thinking about and maybe clarifying in the documentation either way.

Reply:
I went with this based on it seeming like a form of count (which also uses units of 1). I can see the argument though. Given that the output of this as it stands isn't CF compliant I'm not sure there's anything to learn from there.

https://github.com/SciTools/iris/blob/6b273898c3889fd6897351b4c6d4d60660a9397d/lib/iris/analysis/__init__.py#L1692-L1697

Reply:
Yeah, this makes sense and seems consistent with what we have currently. I suppose the use of "data occurences" in each of the aggregator docstrings is enough to clarify what is actually being calculated.